src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageType.cs

```diff
@@ -78,8 +78,12 @@ static Version ParseVersion(ReadOnlySpan<char> input)
         if (versionPrefixIndex >= 0)
         {
             input = input[(versionPrefixIndex + versionPrefix.Length)..];
+
             var firstComma = input.IndexOf(',');
-            input = input[..firstComma];
+            if (firstComma > 0)
```

Comment:
Seems like this should be 

```suggestion
            if (firstComma >= 0)
```

Also, nitpick: I prefer to seen a blank line before an `if`